Core/ModuleInstaller.cs

```diff
@@ -967,7 +967,16 @@ public void AddRemove(ref HashSet<string> possibleConfigOnlyDirs, RegistryManage
         /// </summary>
         public void Upgrade(IEnumerable<string> identifiers, IDownloader netAsyncDownloader, ref HashSet<string> possibleConfigOnlyDirs, RegistryManager registry_manager, bool enforceConsistency = true)
         {
-            var resolver = new RelationshipResolver(identifiers.ToList(), null, RelationshipResolver.DependsOnlyOpts(), registry_manager.registry, ksp.VersionCriteria());
+            // When upgrading, we are removing these mods first and install them again afterwards (but in different versions).
+            // So the list of identifiers of modulesToRemove and modulesToInstall is the same,
+            // RelationshipResolver take care of finding the right CkanModule for each identifier.
+            List<string> identifierList = identifiers.ToList();
+            var resolver = new RelationshipResolver(
+                identifierList,
+                identifierList,
```

Comment:
Maybe we should strip off a `=version` suffix in case the user passes it?

Reply:
Never mind, somehow this already works just fine as-is...

```
PS C:\Users\user\github\CKAN> .\_build\ckan.exe install Kopernicus-BE=UBEE_1101_45
About to install...

 * Kopernicus Bleeding Edge Beta - DEV RELEASE UBEE_1101_45 (cached)
 * Module Manager 4.1.4 (cached)


Continue? [Y/n]


Installing mod "Kopernicus-BE UBEE_1101_45"
Installing mod "ModuleManager 4.1.4"
Updating registry
Committing filesystem changes
Rescanning GameData
Done!

PS C:\Users\user\github\CKAN> .\_build\ckan.exe upgrade Kopernicus-BE=UBEE_1101_44

Upgrading modules...

About to upgrade...

Downloading "https://github.com/R-T-B/Kopernicus/releases/download/UBE-release-44/KopernicusBE_1101_Release44.zip"
0 B/sec - downloading - 0 B left - 100%           6366 [1] WARN CKAN.ModuleInstaller (null) - Downgrading Kopernicus-BE from UBEE_1101_45 to UBEE_1101_44

Removing "CKAN.InstalledModule"
Installing "Kopernicus-BE UBEE_1101_44"
Updating registry
Committing filesystem changes
Done!
Done!

PS C:\Users\user\github\CKAN> .\_build\ckan.exe upgrade Kopernicus-BE=UBEE_1101_46

Upgrading modules...

About to upgrade...

Installing previously uninstalled mod ModularFlightIntegrator

Removing "CKAN.InstalledModule"
Installing "Kopernicus-BE UBEE_1101_46"
Installing "ModularFlightIntegrator 1.2.7.0"
Updating registry
Committing filesystem changes
Done!
Done!
```

Reply:
Oh, I added the sanitization inside `RelationshipResolver.ctor` instead of `Upgrade()`, to also fix it for other possible routes.